.github/workflows/slack-notification.yml

```diff
@@ -0,0 +1,54 @@
+---
+
+name: 'Slack notification'
+
+on:
+  workflow_call:
+    inputs:
+      application:
+        description: 'The friendly name for the application'
+        type: string
+        required: true
+      environment:
+        description: 'The environment that is being targeted for this deployment'
+        type: string
+        required: true
+      version:
+        description: 'The version that is being deployed'
+        type: string
+        default: ${{ github.ref }}
+      type:
+        description: 'The type of action that is being performed such as build or deployment'
+        type: string
+        required: true
+      status:
+        description: 'The status of the deployment.  Options are started, success, and failure'
+        type: string
+        required: true
+
+permissions:
+  contents: read # default
+  id-token: write # for aws auth
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  notify:
+    name: 'Notify'
+    runs-on: ubuntu-latest
+    environment: ${{ inputs.environment }}
```

Comment:
This is not needed here because we should be taking in a secret: `slack_webhook` on the workflow.  We cannot depend on `secrets.SLACK_WEBHOOK_URL` in this case because it would be defined per repo.